spec/registry.md

```diff
@@ -384,6 +322,158 @@ its _resolved value_ contains the implementation-defined integer value
+In general, the default values for such options depend on the locale, 
+the currency,
+the value of other options, or all of these.
+
+Fraction digits for currency values behave differently than for other numeric formatters.
+The number of fraction digits displayed is usually set by the currency used.
+For example, USD uses 2 fraction digits, while JPY uses none.
+Setting some other number of `fractionDigits` allows greater precision display
+(such as when performing currency conversions or other specialized operations)
+or disabling fraction digits if set to `0`.
+The special _option_ _value_ `hideIfWhole` is used to display values without
+fraction digits when the number of fraction digits is zero,
+or based on the currency when the number of fraction digits for the currency is non-zero.
+> For example, this _message_:
+> ```
+> The special price is {$price :currency fractionDigits=hideIfWhole}.
+> ```
+> When used with the value `5.00 USD` in the `en-US` locale displays as:
+> ```
+> The special price is $5.
+> ```
+> But like this when when value is `5.01 USD`:
+> ```
+> The special price is $5.01.
+> ```
+
+Implementations MAY internally alias option values that they do not have data or a backing implementation for.
+Notably, the `currencyDisplay` option has a rich set of values that mirrors developments in CLDR data.
+Some implementations might not be able to produce all of these formats for every currency.
+
+> [!NOTE]
+> Except where noted otherwise, the names of _options_ and their _values_ were derived from the
+> [options](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#options)
+> in JavaScript's `Intl.NumberFormat`.
+
+> [!NOTE]
+> The option `select` does not accept the value `ordinal` because selecting
+> currency values using ordinal rules makes no sense.
+
+The following options and their values are required to be available on the function `:currency`:
+- `select`
+   -  `plural` (default)
+   -  `exact`
+- `currency`
+   - valid [Unicode Currency Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeCurrencyIdentifier)
+     (no default)
+- `compactDisplay` (this option only has meaning when combined with the option `notation=compact`)
+   - `short` (default)
+   - `long`
+- `notation`
+   - `standard` (default)
+   - `compact`
+- `numberingSystem`
+   - valid [Unicode Number System Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeNumberSystemIdentifier)
+     (default is locale-specific)
+- `currencySign`
+  - `accounting`
+  - `standard` (default)
+- `currencyDisplay` (this option's values are derived from those in ICU [NumberFormatter.UnitWidth](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/number/NumberFormatter.UnitWidth.html))
+  - `narrow`
+  - `short` (default)
+  - `full`
+  - `iso`
+  - `formal`
+  - `variant`
+  - `none` (this is called `hidden` in ICU)
```

Comment:
I continue to object to using a different source for this one single function option, when all the other options for all the other functions are using TC39 Intl as their source. Using this set of option values unnecessarily breaks from our otherwise well established patterns.

Reply:
We discussed this in yesterday's call (which you weren't able to attend). The challenge here is that currency formatting has been greatly improved in CLDR (which is reflected by the changes in ICU) after `Intl.NumberFormat` was created. Those changes have not yet materialized in TC39, but really are needed. FWIW, there is continuing evolution here. To wit, there are a number of currencies with more than one symbol (some of which are `formal` and some of which are `variant`). These symbols need to be selectable and, when selected, can result in different formatting patterns.

The poster child for this is the Turkish Lira, which has a newfangled symbol, which displays on the left thusly: `₺123.456,79` , but which is very often rendered using the older ASCII symbol "TL":

![image](https://github.com/user-attachments/assets/b6af582b-b290-42fc-b3bc-38a612f7f811)

The discussion yesterday recognized that there are more options here than in Intl, but that implementations are only required to accept all these keywords--not to provide different results based on them. Intl provides `code`, `symbol`, `narrowSymbol`, and `name`. We could reasonably rename thusly:

| Proposed Above | Intl Equivalent | Suggested | Comments |
|:---:|:---:|:---:|---|
| `narrow` | `narrowSymbol` | `narrow` | No need to be extra verbose |
| `short` | `symbol` | `symbol` | (default) `symbol` is consistent with user expectations |
| `full` | `name` | `name` | ICU's name is actually `FULL_NAME`. Can be misleading, e.g. JPY renders as `123,457円` |
| `iso` | `code` | `code` | `code` might be better because some codes are not ISO4217 |
| `formal` | n/a | `formal` | Variations such as `TRD` => `NT$` |
| `variant` | n/a | `variant` | Variations such as the `TRY` one illustrated above |
| `none` | n/a | `none` | Frequently used for columns of currency values |

Would that rename work for you?

Reply:
I did the rename just now (in bbc7091) but am looking for @eemeli's feedback